### PR TITLE
Added Optional React

### PR DESCRIPTION
**Pull request by sayatul**
[FP-5159]

### CODE REVIEW

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/index.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta http-equiv="Content-type" content="text/html; charset=utf-8"/>
+  <link rel="stylesheet" href="https://sdks.shopifycdn.com/polaris/3.4.0/polaris.min.css" />    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-5sAR7xN1Nv6T6+dT2mhtzEpVJvfS3NScPQTrOxhwjIuvcA67KV2R5Jz6kr4abQsz" crossorigin="anonymous">
```
Comment (tabrez96) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/index.html]: "link"s in separate lines

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        userMenu={userMenuMarkup}
+        onNavigationToggle={this.toggleState('showMobileNavigation')}
+      />
+    );
+
+     return topBarMarkup;
+  }
+
+  renderNavigation() {
+    const { userData, navItemCount, onNavItemClick, refreshCount } = this.props;
```
Comment (tabrez96) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: this line not needed

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+          onNavigationDismiss={this.toggleState('showMobileNavigation')}
+        >
+        </Frame>
+      </AppProvider>
+    )
+  }
+}
+
+
+FrameComponent.propTypes = {
```
Comment (tabrez96) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: not needed for now

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    );
+
+    const navigationMarkup = (
+      <Navigation location="/" userMenu={navigationUserMenuMarkup}>
+        <Navigation.Section
+          items={[
+            {
+              label: 'Dashboard',
+              icon: 'home',
+              onClick: () => { onNavItemClick('base.dashboard.all') }
```
Comment (tabrez96) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: Change this

File: {{cookiecutter.app_name}}/{% if cookiecutter.use_react == "yes" %}package.json{% endif %} [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "{{cookiecutter.app_name}}",
+  "version": "1.0.0",
+  "description": "{{cookiecutter.project_short_description}}",
+  "main": "webpack.config.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "dev": "webpack --progress --watch"
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{% if cookiecutter.use_react == "yes" %}package.json{% endif %}]: add command for production also `build`

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/api/index.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    return JSON.parse(responseData)
+  }]
+});
+
+// Handle 401
+instance.interceptors.response.use( (response) => {
+  return response;
+}, (error) => {
+  if (error.response.status === 401) {
+    // TODO: Change to react component
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/api/index.js]: ?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/api/index.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    return window.location = '/login'
+  } else {
+    return Promise.reject(error);
+  }
+})
+
+const Api = {
+  user: () => instance.get("/user", {data: null}),
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/api/index.js:24]: why data null? Put a comment

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/components/popover/index.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import PropTypes from 'prop-types';
+
+import { ActionList, Popover } from '@shopify/polaris';
+
+class PopoverComponent extends React.PureComponent {
+ constructor(props) {
+   super(props);
+   this.state = {};
+
+    this.togglePopover = this.togglePopover.bind(this);
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/components/popover/index.js]: indent?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+const theme = {
+  colors: {
+    topBar: {
+      background: '#1e3d58',
+    },
+  },
+  logo: {
+    width: 140,
+    topBarSource:
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js:17]: move this to styles.css or some relevant file

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+      this.setState({
+        userName: res.data.name
+      })
+    })
+
+    this.userMenuActions = [
+      {
+        items: [
+          {content: 'Logout', icon: 'logOut',
+            onAction: () => { this.onLogout() }
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js:38]: why on `componentDidMount `

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+   renderTopBar() {
+    const { userMenuOpen, userName } = this.state;
+
+    var initials = userName.match(/\b\w/g) || [];
+    initials = ((initials.shift() || '') + (initials.pop() || '')).toUpperCase();
+
+    const userMenuMarkup = (
+      <TopBar.UserMenu
+        actions={this.userMenuActions}
+        name={'Atul Yadav'}
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: ?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+      />
+    );
+
+     return topBarMarkup;
+  }
+
+  renderNavigation() {
+    const { userName } = this.state;
+
+    var initials = userName.match(/\b\w/g) || [];
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: move this to a function

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/user/{% if cookiecutter.use_react == "yes" %}views.py{% endif %} [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+    return send_from_directory("static", "react/dist/index.html")
+
+
+@blueprint.route('/user')
+@login_required
+def user_details():
+    return jsonify(**oauth_user.serialize)
+
+
+@blueprint.route('/login', methods=['GET', 'POST'])
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/user/{% if cookiecutter.use_react == "yes" %}views.py{% endif %}]: POST?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/utils.py [hunk 14]
```diff
@@ -15,6 +15,8 @@ def login_required(f):
     @wraps(f)
     def decorated_function(*args, **kwargs):
         if flask.session.get('fulfil') is None:
+            if flask.request.is_xhr or flask.request.is_json:
+                flask.abort(401)
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/utils.py]: import explicitly

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 15]
```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+ import { AppProvider, Frame, ActionList, Card, TopBar, Navigation, Link, Button } from '@shopify/polaris';
+import { Icon } from 'octantis';
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: used where?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+import * as _ from 'lodash';
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: please remove unused deps

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+import Popover from '../../components/popover/index.js';
+import Api from '../../api/index.js';
+
+import style from './style.css';
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js:6]: used where?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+    super(props);
+    this.state = {
+      userName: ''
+    };
+    this.userMenuActions = [];
+
+  }
+
+  componentDidMount() {
+    Api.user().then((res) => {
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js]: error handling?

File: {{cookiecutter.app_name}}/{% if cookiecutter.use_react == "yes" %}package.json{% endif %} [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "author": "{{cookiecutter.full_name}}",
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{% if cookiecutter.use_react == "yes" %}package.json{% endif %}]: it should be email I guess

**Review by Anmol-Gulati — CHANGES_REQUESTED** [review]

**Comment by Anmol-Gulati** [comment]
@sayatul Please add react client build to heroku release command

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js (hunk 9 above)
Comment (sayatul) [reply {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/static/{% if cookiecutter.use_react == "yes" %}react{% endif %}/src/app/containers/dashboard/index.js:17]: 1) Shopify gives provision to add images for logo in frame like this.
2) Frame doesn't accept 'className' props

**Pull request merged**